code/game/objects/items/devices/lightreplacer.dm

```diff
@@ -142,10 +142,14 @@
 
 		to_chat(user, span_notice("You fill \the [src] with lights from \the [S]. " + status_string() + ""))
 
-/obj/item/lightreplacer/emag_act()
+/obj/item/lightreplacer/emag_act(mob/user, obj/item/card/emag/emag_card)
 	if(obj_flags & EMAGGED)
-		return
-	Emag()
+		return FALSE
+	obj_flags |= EMAGGED
+	name = "shortcircuited [initial(name)]"
```

Comment:
This should be in ``update_name`` and the update appearance below should remove the ``UPDATE_ICON`` arg so it updates